alacritty_terminal/src/term/mod.rs

```diff
@@ -2128,14 +2107,48 @@ impl<T: EventListener> Handler for Term<T> {
         self.cursor_style = style;
     }
 
+    #[inline]
+    #[cfg(not(windows))]
+    fn set_title(&mut self, title: &str) {
+        if self.dynamic_title {
+            trace!("Setting window title to '{}'", title);
+
+            self.title = Some(title.into());
+            self.event_proxy.send_event(Event::Title(title.to_owned()));
+        }
+    }
+
+    #[inline]
+    #[cfg(windows)]
+    fn set_title(&mut self, title: &str) {
+        if self.dynamic_title {
+            // cmd.exe in winpty: winpty incorrectly sets the title to ' ' instead of
+            // 'Alacritty' - thus we have to substitute this back to get equivalent
+            // behaviour as conpty.
+            //
+            // The starts_with check is necessary because other shells e.g. bash set a
+            // different title and don't need Alacritty prepended.
+            trace!("Setting window title to '{}'", title);
+
+            let title = if !tty::is_conpty() && title.starts_with(' ') {
+                format!("Alacritty {}", title.trim())
+            } else {
+                title.to_owned()
+            };
+
+            self.title = Some(title.clone());
+            self.event_proxy.send_event(Event::Title(title));
+        }
+    }
```

Comment:
I'm questioning if this separation of Windows/Unix is really necessary here and what exactly the benefits of `set_title`'s Windows implementation are.

So potentially we can just run `title.trim()` on all platforms and unify the code? If we change that, I think it makes sense to look into that now while already touching the code anyways.

Reply:
well we need some Windows testing then. I'm always in favor of having less platform specific logic.

Reply:
@davidhewitt Can you help with this maybe, since you wrote the original comment?

The comment says that this is about WinPTY setting the title to just ` `, but what we do is *always* prepending `Alacritty `, which seems wrong if the comment is correct? It should be more like `if title == ' ' && winpty { "Alacritty" }`, right?

Also I'm not quite sure under which circumstances the title is set to ` `, does ConPTY reset it to what was initially passed to it at startup? What if the title isn't set in the ConPTY code, does it also set it to ` ` then?

Reply:
Yes - the 'title' as far as cmd.exe is concerned is a parameter passed at startup:

https://github.com/alacritty/alacritty/blob/bb0a214a157adc81d1e598602bba7739ecec8672/alacritty_terminal/src/tty/windows/conpty.rs#L143

I'm not sure exactly what cmd.exe's logic for the title is, but it seems to be showing _either_ this startup title plus the cwd, or whatever the hosted application has set the title to.

WinPTY doesn't let us pass this title at startup, instead hard-coding it to ` `. So that's what this code is trying to work around.

In all honesty I think this code can be binned; Alacritty will then behave consistently on all platforms and even though WinPTY + dynamic_title will be slightly "broken", it's a cosmetic issue that will affect a tiny fraction of users.

Reply:
@davidhewitt So unless the title is reset on WinPTY, it will always be reasonable and we *shouldn't* change anything about it? So our current logic would be incorrect when the title is set to something that coincidentally starts with some whitespace?

If we don't set the title on ConPTY, should every thing still work out fine just using `Alacritty.exe` as title instead of `Alacritty`?